cody.el

```diff
@@ -67,11 +68,114 @@ This is generated and cached on first use, if telemetry is enabled."
 (defcustom cody-enable-automatic-completions t
   "Non-nil to have Cody prompt with suggestions automatically.
 Completion suggestions will appear after you stop typing for a while,
-if any suggestions are available. Set this to nil if you prefer to
+if any suggestions are available.  Set this to nil if you prefer to
 use manual completion triggering with `cody-request-completion'."
   :type 'boolean
   :group 'cody)
 
+;;; Completions "API" for the rest of the lisp code:
+
+(defclass cody-completion-item ()
+  ((insertText :initarg :insertText
+               :initform nil
+               :type (or null string)
+               :documentation "The original suggested text.")
+   (range :initarg :range
+          :initform nil
+          :type (or null list)
+          :documentation "The range where the completion applies.")
+   (-displayText :initarg :displayText
+                :initform nil
+                :type (or null string)
+                :documentation "The updated text as the user types."))
+  "Represents a single alternative/suggestion in a completion response.")
+
+(cl-defmethod cody-get-display-text ((item cody-completion-item))
+  "Get the display text for the completion item ITEM."
+  (oref item -displayText))
+
+(cl-defmethod cody-set-display-text ((item cody-completion-item) value)
+  "Set the display text for the completion item ITEM to VALUE."
+  (oset item -displayText value))
+
+(defclass cody-completion ()
+  ((items :initarg :items
+          :initform nil
+          :type (or null vector)
+          :accessor cody--completion-items
+          :documentation "Vector of `cody-completion-item' instances.")
+   (completionEvent :initarg :completionEvent
+                    :initform nil
+                    :type (or null list)
+                    :documentation "Event associated with the completion.")
+   (response :initarg :response
+             :initform nil
+             :type (or null list)
+             :accessor cody--completion-response
+             :documentation "Top-level jsonrpc protocol response object.")
+   (-current-index :initform 0
```

Comment:
what is the significance of the `-` prefix? My guess is it is convention for "private" but I may be completely off the mark.

Reply:
Yep, private.